File: src/splash/java/dev/su5ed/koremods/api/SplashScreen.java
package dev.su5ed.koremods.api;

import org.apache.logging.log4j.core.LoggerContext;

public interface SplashScreen {
    void startThread();
    
    void awaitInit();
    
    void setTerminateOnClose(boolean terminate);
    
    void injectSplashLogger(LoggerContext context);
    
    void log(String message);
    
    void close(boolean delay);
    
    boolean isClosing();
}

File: src/splash/kotlin/dev/su5ed/koremods/api/SplashScreen.kt
package dev.su5ed.koremods.api

import org.apache.logging.log4j.core.LoggerContext

interface SplashScreen {
    fun awaitInit()
    
    fun injectSplashLogger(context: LoggerContext)
    
    fun log(message: String)
    
    fun close()
    
    fun isClosing(): Boolean
}
